runtime/runtime/src/lib.rs

```diff
@@ -1651,6 +1643,20 @@ impl Runtime {
         state_update.commit(StateChangeCause::UpdatedDelayedReceipts);
         self.apply_state_patch(&mut state_update, state_patch);
         let (trie, trie_changes, state_changes) = state_update.finalize()?;
+        if let Some(prefetcher) = &prefetcher {
+            // Only clear the prefetcher queue after finalize is done because as part of receipt
+            // processing we also prefetch account data and access keys that are accessed in
+            // finalize. This data can take a very long time otherwise if not prefetched.
+            //
+            // (This probably results in more data being accessed than strictly necessary and
+            // prefetcher may touch data that is no longer relevant as a result but...)
```

Comment:
IIUC we check compute limit as we go through the delayed receipts and the prefetching happens inside the loop:

https://github.com/near/nearcore/blob/1d082db0cb803de917c92ed7f10b365f4f5a7ad5/runtime/runtime/src/lib.rs#L1487-L1507

Reply:
The downside of that is there is very little delay between when the prefetcher is invoked for the receipt and when the receipt is actually processed. I wonder if there are some gains which could be had here that would improve throughput for delayed receipts.

Could it be worth trying:

1. Implement a new prefetcher method which chains `prefetch_trie_key` into `prefetch_receipts_data`. We could call it with a trie key `DelayedReceipt { index }` and basically it would understand that the trie value is expected to be a receipt and it should perform prefetching for the receipt as well. 
2. Call the new prefetcher method on the next index in the delayed receipts queue while we are processing the current one. Or in general, have some sliding window of K indices over which we invoke the prefetcher.

Reply:
The downside of that is there is very little delay between when the prefetcher is invoked for the receipt and when the receipt is actually processed.

Could it be worth trying:

1. Implement a new prefetcher method which chains `prefetch_trie_key` into `prefetch_receipts_data`. We could call it with a trie key `DelayedReceipt { index }` and basically it would understand that the trie value is expected to be a receipt and it should perform prefetching for the receipt as well. 
2. Call the new prefetcher method on the next index in the delayed receipts queue while we are processing the current one. Or in general, have some sliding window of K indices over which we invoke the prefetcher.

Reply:
I don't think we need to do anything as fancy. We can take a guess that we fit e.g. "about 500 delayed receipts" and start prefetching them even before we start executing transactions at the beginning of the `apply`. Even if we are overzealous with how many we do end up prefetching for that specific chunk, its not like those receipts are going anywhere -- we'll have to process them next iteration around anyway. And at that point the receipts would still be in some of the caches, so they would be faster to prefetch on the next iteration.

It is true that we prefetch them just before they are necessary currently, and at least some of the metrics do indicate that before the `clear()` call was moved to-after-finalize we would "cancel" some prefetch requests quite consistently (suggesting that we indeed prefetch a little too Just In Time.)

Regardless these are all good thoughts to evaluate if and when we improve on this code. I don't think we strictly need to add to this PR to address anything here so far.